packages/ember-intl/addon/services/intl.ts

```diff
@@ -0,0 +1,397 @@
+import { getOwner } from '@ember/application';
+import { registerDestructor } from '@ember/destroyable';
+import { cancel, next, type Timer as EmberRunTimer } from '@ember/runloop';
+import Service from '@ember/service';
+import { htmlSafe } from '@ember/template';
+import { tracked } from '@glimmer/tracking';
+import EventEmitter from 'eventemitter3';
+
+import type {
+  FormatDateParameters,
+  FormatListParameters,
+  FormatMessageParameters,
+  FormatNumberParameters,
+  FormatRelativeParameters,
+  FormatTimeParameters,
+  IntlShape,
+  OnErrorFn,
+} from '../-private/formatjs/index';
+import {
+  createIntl,
+  createIntlCache,
+  formatDate,
+  formatList,
+  formatMessage,
+  formatNumber,
+  formatRelative,
+  formatTime,
+} from '../-private/formatjs/index';
+import { escapeFormatMessageOptions } from '../-private/utils/escape-format-message-options';
+import { getDOM } from '../-private/utils/get-dom';
+import {
+  convertToArray,
+  convertToString,
+  hasLocaleChanged,
+} from '../-private/utils/locale';
+import { flattenKeys, type Translations } from '../-private/utils/translations';
+import translations from '../translations';
+
+type OnFormatjsError = (error: Parameters<OnErrorFn>[0]) => void;
+
+export default class IntlService extends Service {
+  @tracked private _intls: Record<string, IntlShape> = {};
+  @tracked private _locale?: string[];
```

Comment:
**breaking**

The `locale` getter (a copy of `_locale)` and setter no longer exist. End-developers are to use `setLocale()` to update `_locale`.